modules/ui/fb_feature_picker.js

```diff
@@ -13,36 +13,29 @@ import { uiRapidFirstEdit } from './rapid_first_edit_dialog';
 
 export function uiFbFeaturePicker(context, keybinding) {
     var _datum;
-    var ML_ROADS_LIMIT_NON_TM_MODE = 50;
+    var AI_FEATURES_LIMIT_NON_TM_MODE = 50;
 
 
-    function isAddRoadDisabled() {
+    function isAddFeatureDisabled() {
         // when task GPX is set in URL (TM mode), "add roads" is always enabled
         var gpxInUrl = utilStringQs(window.location.hash).gpx;
         if (gpxInUrl) return false;
 
-        var mlRoadsCount = 0;
-        var entities = context.graph().entities;
-        for (var eid in entities) {
-            var e = entities[eid];
-            if (eid.startsWith('w-') && e && (e.tags.source === 'digitalglobe' || e.tags.source === 'maxar')) {
-                mlRoadsCount += 1;
-            }
-        }
-        return mlRoadsCount >= ML_ROADS_LIMIT_NON_TM_MODE;
+        var numAiFeatures = context.history().countAIFeaturesAdded(); 
```

Comment:
Let's keep this logic in this file, with the strategy of modifying iD's internals as unintrusively as we can. To do so, this can call `context.history().peekAllAnnotations()` and then iterate to count.

Reply:
Good call! history.js is now untouched by this change.